sw/device/tests/rv_timer_test.c

```diff
@@ -32,28 +43,58 @@ bool test_main(void) {
 
   irq_global_ctrl(true);
   irq_timer_ctrl(true);
-  rv_timer_intr_enable(hart, true);
 
-  intr_handling_success = false;
-  rv_timer_set_us_tick(hart);
-  rv_timer_set_cmp(hart, cmp);
-  rv_timer_ctrl(hart, true);
+  mmio_region_t timer_reg =
+      mmio_region_from_addr(TOP_EARLGREY_RV_TIMER_BASE_ADDR);
+  CHECKZ(dif_rv_timer_init(
```

Comment:
I agree with @gkelly  (don't exactly remember which PR he has mentioned it), could you please use `CHECK` instead?

I am not a big fan of how `clang-format` formats function calls inlined in `CHECK` and `CHEKZ`. Please consider using:
```
some_type_t result = function_call(...);
CHECK(result == kDif...Ok, "some debug output");
```

Reply:
CHECKZ is gone; I deleted it from `check.h`.

Regarding your code example; while clang-format doesn't produce the nicest code, it's fine the way it is. In particular, `CHECK` without a debug message will now print the source of the expression it was checking, so for a lot of code it makes a lot more sense to write `CHECK(expr)`, which is less redundant.

If the `== Ok` is hurting line length, we can probably add a `CHECK_OK` variant, but I think that's unnecessary.

Reply:
Actually, https://abseil.io/tips/161 captures why doing writing CHECK(result) is a bad idea. This isn't a complicated expression; we just have really long names for our functions because C.

(Yes, this is a C++ tip, but a lot of the Abseil ToTWs are very relevant to writing good C, too.)

Reply:
This is a bit unfortunate, because `assert(some_side_effecting_call())` *is* different from `result = some_side_effecting_call(); assert(result)` (because of NDEBUG).

C programmers will hopefully have developed an aversion to `assert(foo())`: we now require them to remember that `CHECK(foo())` is fine (and preferred).